test/compare-acorn-es2020/optional-chaining.expected.js

```diff
@@ -0,0 +1,16 @@
+obj.aaa.bbb;
+obj.aaa?.bbb;
+obj?.aaa.bbb;
+obj?.aaa?.bbb;
+obj.aaa.bbb;
+obj.aaa?.bbb;
+obj?.aaa.bbb;
```

Comment:
This should be `(obj?.aaa).bbb`. I don't trust these expectations. Can you add an assertion to the tests that the result of parsing the output of the codegen is the same as the result of parsing the original input?

Reply:
You are right. But i'm not good with parenthesis. Anybody can help?

Reply:
@michaelficarra updated. Now expected results same as `babel` results.
I updated logic, but not sure that did it right, but all test are passed now.

Reply:
@sanex3339 You've not added the round-tripping assertion I asked for.

Reply:
@michaelficarra in which test file i can add it?

Reply:
I would put it right [here](https://github.com/estools/escodegen/pull/412/files#diff-cacb5c391de59a37897c31fde28d1783R49). And the same for `testMin`.

Reply:
Hmm. Looks like AST tree of the output cannot be 1 to 1 matched with input AST tree, because of the parens and how AST tree will be generated.

But the generated code is the same as generated code from the babel. 
Tried it here:
https://babeljs.io/repl/#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=PYIwVgdAhjEvBuAUKSMoH46JeL7sRlV8Yt4ikAKVaGASkORvDqnvJxbFPaepJsO_Kt16MKEAMYzGAEwXMxQwlhlT5igXiGdKqANroAugYrHi4IzGNYz8C7h4RrUU-cvPXtlx-1hventHbkC7P1CTYXsQEMFAmLidMN8HJUMoxIN1YyCFORDlBPM7HLyFRyA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&timeTravel=false&sourceType=module&lineWrap=false&presets=&prettier=false&targets=&version=7.10.2&externalPlugins=babel-plugin-minify-dead-code-elimination%400.5.1

Reply:
that is wat i'm talking about:
![image](https://user-images.githubusercontent.com/9210534/84543317-d5b2e680-ad03-11ea-8342-6575b639416b.png)

Reply:
@michaelficarra beat it! Added AST comparison tests. Now AST is the same, but the output code is different from Babel output, because we have to keep some parenthesis to make input AST the same as output AST